chat-commands.js

```diff
@@ -2479,6 +2479,59 @@ exports.commands = {
 	},
 	blacklisthelp: [`/blacklist [username], [reason] - Blacklists the user from the room you are in for a year. Requires: # & ~`],
 
+	bb: 'battleban',
+	battleban: function (target, room, user) {
+		if (!target) return this.parse('/help battleban');
+
+		let reason = this.splitTarget(target, true);
+		let targetUser = this.targetUser;
+
+		if (!targetUser) {
+			return this.errorReply(`User '${this.targetUsername}' not found.`);
+		}
+		if (!this.can('forcerename', targetUser)) return false;
+		if (targetUser.battleBanned) return this.errorReply(`User '${targetUser.name}' is already banned from battles.`);
+
+		const reasonText = reason ? ` (${reason})` : `.`;
+		const battlebanMessage = `${targetUser.name} was banned from starting new games by ${user.name}${reasonText}`;
+		this.privateModAction(`(${battlebanMessage})`);
+
+		// Notify staff room when a user is locked outside of it.
+		if (room.id !== 'staff' && Rooms('staff')) {
+			Rooms('staff').addByUser(user, "<<" + room.id + ">> " + battlebanMessage);
+		}
+
+		this.globalModlog("BATTLEBANNED", targetUser, ` by ${user.userid}${reasonText}`);
+		Ladders.cancelSearches(targetUser);
+		Punishments.battleban(targetUser, null, null, reason);
+		targetUser.popup(`|modal|${user.name} has prevented you from starting new battles${reasonText}`);
+		return true;
+	},
+	battlebanhelp: [`/battleban OR /bb [username], [reason] - Prevents the user from started new battles and shows them the [reason]. Requires: % @ * & ~`],
+
+	ubb: 'unbattleban',
+	unbattleban: function (target, room, user) {
+		if (!target) return this.parse('/help unbattleban');
+		if (!this.can('forcerename')) return false;
+
+		let targetUser = Users.get(target);
+		let reason = '';
+		if (targetUser && targetUser.battleBanned) {
+			reason = ' (' + targetUser.battleBanned + ')';
+		}
+
+		let letplay = Punishments.unbattleban(target);
```

Comment:
This is a pretty bad variable name, what about ``const unbanned`` or ``const canBattle``?

Reply:
Speaking of which, we should probably introduce a ``this.canBattle`` function if this feature does happen